common/src/main/java/com/lightstep/opentelemetry/common/VariablesConverter.java

```diff
@@ -70,53 +69,84 @@ public static void setSystemProperties(String spanEndpoint,
+    System.setProperty("otel.exporter.otlp.span.endpoint", configuration.spanEndpoint);
+    System
+        .setProperty("otel.exporter.otlp.span.insecure",
+            String.valueOf(configuration.insecureTransport));
+    System
+        .setProperty("otel.exporter.otlp.span.timeout", String.valueOf(DEFAULT_LS_DEADLINE_MILLIS));
+    System.setProperty("otel.exporter.otlp.span.headers",
+        "lightstep-access-token=" + configuration.accessToken);
+    if (configuration.propagators != null) {
+      System.setProperty("otel.propagators", configuration.propagators);
     }
 
-    String otelResourceAttributes = "service.name=" + serviceName;
-    if (serviceVersion != null) {
-      otelResourceAttributes += ",service.version=" + serviceVersion;
+    String otelResourceAttributes = "service.name=" + configuration.serviceName;
+    if (configuration.serviceVersion != null) {
+      otelResourceAttributes += ",service.version=" + configuration.serviceVersion;
     }
 
-    if (resourceAttributes == null || resourceAttributes.isEmpty()
-        || !resourceAttributes.contains("host.name=")) {
+    if (configuration.resourceAttributes == null || configuration.resourceAttributes.isEmpty()
+        || !configuration.resourceAttributes.contains("host.name=")) {
       String hostname = getHostName();
       if (hostname != null && !hostname.isEmpty()) {
         otelResourceAttributes += ",host.name=" + hostname;
       }
     }
 
-    if (resourceAttributes != null && !resourceAttributes.isEmpty()) {
-      otelResourceAttributes += "," + resourceAttributes;
+    if (configuration.resourceAttributes != null && !configuration.resourceAttributes.isEmpty()) {
+      otelResourceAttributes += "," + configuration.resourceAttributes;
     }
 
     System.setProperty("otel.resource.attributes", otelResourceAttributes);
 
-    if (logLevel != null) {
-      System.setProperty("io.opentelemetry.javaagent.slf4j.simpleLogger.defaultLogLevel", logLevel);
-      if (logLevel.equals("debug")) {
-        String msg = "spanEndpoint: " + spanEndpoint;
-        if (propagators != null) {
-          msg += ", propagators: " + propagators;
+    if (configuration.logLevel != null) {
+      System.setProperty("io.opentelemetry.javaagent.slf4j.simpleLogger.defaultLogLevel",
+          configuration.logLevel);
+      if (configuration.logLevel.equals("debug")) {
+        String msg = "spanEndpoint: " + configuration.spanEndpoint;
+        if (configuration.propagators != null) {
+          msg += ", propagators: " + configuration.propagators;
         }
-        msg += ", accessToken: " + accessToken + ", serviceName: " + serviceName;
+        msg += ", accessToken: " + configuration.accessToken + ", serviceName: "
+            + configuration.serviceName;
         logger.info(msg);
       }
     }
 
-    // Do not enable metrics
-    System.setProperty("otel.exporter", "otlp_span");
+    if (configuration.metricsEnabled) {
```

Comment:
I'm thinking about a follow-up later, so we can have individual methods for each section. Problem is, this method is getting _too_ big ;)

Reply:
yeah, but it is in https://github.com/lightstep/specs/blob/main/daq/launcher.md

Reply:
Sounds fine.